drivers/hmis/app/models/hmis/ce/referral_message_handler.rb

```diff
@@ -58,9 +57,26 @@ def start_referral
       referral.opportunity.reserve!
     end
 
-    def accept_referral
+    def accept_referral(message)
       referral.accept!
       referral.opportunity.close!
+
+      referral.workflow_instance.audit_events.create!(
+        event_type: 'end_workflow',
+        user: message.user,
+        event_data: { 'result': 'accept_referral' }.to_h,
```

Comment:
This is a new data shape in the `event_data` field, which the schema object reads from to determine what event text to show the user. Let's discuss -- I wasn't sure about the intended api of this `event_data` json blob or how locked down we wanted to keep it.

Reply:
This makes sense to me. I've put the constants in the engine, since they are related to referral workflow execution. But they aren't actually used there, and could definitely live elsewhere. Let me know if you prefer another place for them.